### PR TITLE
feat: support changeset releases

### PR DESCRIPTION
**Pull request by wonderfulfrog**
## Description

This PR adds support for `changesets` when cutting new releases. I also updated the GH actions to support this (following the example of react-modal).

## Notes

This is my first time adding the `changesets` feature, so please let me know if I did anything wrong!

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js 18.x # TODO: Update to 20 once TS changes are in.
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
```
Comment (marlonmarcello) [inline .github/workflows/lint.yml:20]:
```suggestion
      - name: Checkout Repo
        uses: actions/checkout@v4

      - name: Setup Node.js
        uses: actions/setup-node@v4
        with:
          node-version-file: ".nvmrc"
```

This uses the nvmrc so we don't have to worry about mismatched versions

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js 18.x # TODO: Update to 20 once TS changes are in.
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
```
Comment (marlonmarcello) [inline .github/workflows/release.yml:25]:
```suggestion
      - name: Checkout Repo
        uses: actions/checkout@v4

      - name: Setup Node.js
        uses: actions/setup-node@v4
        with:
          node-version-file: ".nvmrc"
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+      - name: Create Release Pull Request or Publish to npm
+        id: changesets
+        uses: changesets/action@v1
+        with:
+          version: npm run version
+          publish: npm run release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (marlonmarcello) [inline .github/workflows/release.yml:41]: I'll add this to the repo for us.

File: package.json [hunk 4]
```diff
@@ -82,6 +86,7 @@
     "react-dom": "17 - 18"
   },
   "dependencies": {
+    "@changesets/cli": "^2.27.7",
```
Comment (marlonmarcello) [inline package.json:89]:
```suggestion
    "@changesets/cli": "^2.27.7",
    "@changesets/changelog-github": "^0.5.0",
```

Comment (marlonmarcello) [reply package.json:89]: This ensures the changelog works with GH.

File: .changeset/config.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+  "$schema": "https://unpkg.com/@changesets/config@3.0.2/schema.json",
+  "changelog": [
+    "@changesets/changelog-github",
+    {
+      "repo": "@wethegit/react-marquee"
+    }
+  ],
```
Comment (marlonmarcello) [inline .changeset/config.json:8]: For this to work we need `@changesets/changelog-github`. I added as a suggestion bellow

**Review by marlonmarcello — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (wonderfulfrog) [reply .github/workflows/lint.yml:20]: Amazing, thank you!

File: .github/workflows/release.yml (hunk 3 above)
Comment (wonderfulfrog) [reply .github/workflows/release.yml:41]: @marlonmarcello Let me know when that's done and I'll mark this as resolved. Thank you for doing that!

Comment (marlonmarcello) [reply .github/workflows/release.yml:41]: Done!

**Review by marlonmarcello — APPROVED** [review]

**Pull request merged**